primitives/src/parachain.rs

```diff
@@ -328,6 +330,39 @@ impl AttestedCandidate {
 	}
 }
 
+/// A fee schedule for messages. This is a linear function in the number of bytes of a message.
+#[derive(PartialEq, Eq, PartialOrd, Hash, Default, Clone, Copy, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+pub struct FeeSchedule {
+	/// The base fee charged for all messages.
+	pub base: Balance,
+	/// The per-byte fee charged on top of that.
+	pub per_byte: Balance,
+}
+
+impl FeeSchedule {
+	/// Compute the fee for a message of given size.
+	pub fn compute_fee(&self, n_bytes: usize) -> Balance {
+		use rstd::mem;
+		debug_assert!(mem::size_of::<Balance>() >= mem::size_of::<usize>());
+
+		let n_bytes = n_bytes as Balance;
+		self.base.saturating_add(n_bytes.saturating_mul(self.per_byte))
+	}
+}
+
+/// Current Status of a parachain.
+#[derive(PartialEq, Eq, Clone, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+pub struct Status {
```

Comment:
why not `#[derive(Default)]`?

Reply:
What should the default be?

Reply:
I was thinking the following:

```rust
Status {
			head_data: HeadData(Vec::default()),
			balance: 0,
			fee_schedule: FeeSchedule {
				base: 0,
				per_byte: 0,
			}
		}
```

but nvm `HeadData` doesn't implement default

Reply:
I don't think there should be a default. It varies per parachain.